### PR TITLE
test: Compare trimmed toml output

### PR DESCRIPTION
**Pull request by jan-kolarik**
Ignore stray newlines from the `toml` file as `toml::format` output has changed for `toml` library versions between fedora-35 and fedora-36.

### CODE REVIEW

File: test/libdnf/system/test_state.cpp [hunk 1]
```diff
@@ -63,6 +63,16 @@ enabled_stream = "stream-1"
 
 )"""};
 
+static inline void erase_newlines(std::string & str) {
+    str.erase(std::remove(str.begin(), str.end(), '\n'), str.cend());
```
Comment (lukash) [inline test/libdnf/system/test_state.cpp:67]: Does this remove all newlines? You don't want to do that... The previous version was better, though I'd only trim the trailing newlines, as this is a workaround and ideally you don't want any stray newlines at either end, the extraneous newline should actually ideally be fixed in `toml11`... (I haven't reported a bug for it yet)

File: test/libdnf/system/test_state.cpp [hunk 2]
```diff
@@ -63,6 +63,16 @@ enabled_stream = "stream-1"
...
+}
+
+static void assert_equal_without_newlines(std::string expected, std::string actual) {
+    erase_newlines(expected);
+    erase_newlines(actual);
```
Comment (lukash) [inline test/libdnf/system/test_state.cpp:72]: You only really need to remove the trailing newline from the stored tomls, not from the static strings that we compare against (or store in the test files).

File: test/libdnf/system/test_state.cpp [hunk 3]
```diff
@@ -63,6 +63,16 @@ enabled_stream = "stream-1"
...
+
+    CPPUNIT_ASSERT_EQUAL(expected, actual);
```
Comment (lukash) [inline test/libdnf/system/test_state.cpp:74]: The issue with doing this is that whenever you get a failure, you'll get the line number of this `CPPUNIT_ASSERT_EQUAL()` call, and you'll be quite clueless which one is actually failing.

I'd return a trimmed copy from a `trimmed(...)` function and call that function on the file contents inside all the ASSERTs.

File: test/libdnf/system/test_state.cpp (hunk 1 above)
Comment (jan-kolarik) [reply test/libdnf/system/test_state.cpp:67]: OK I wanted to simplify things here. The problem is there is/is not a trailing newline between each section, so for the `modules_contents` case, the trailing remove is not enough.

**Pull request merged**